File: Kerberos_Client/Kerberos_Client/Struct/MyStruct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kerberos_Client
{
    public class MyStruct
    {
        #region 证书
        public class Certificate
        {
            string version;//版本号
            string serial;//序列号
            string deadline;//有效日期
            string name;//主体名
            string pk;//公钥
            public string Version
            {
                get { return version; }
                set { version = value; }
            }
            public string Serial
            {
                get { return serial; }
                set { serial = value; }
            }
            public string Deadline
            {
                get { return deadline; }
                set { deadline = value; }
            }
            public string Name
            {
                get { return name; }
                set { name = value; }
            }
            public string Pk
            {
                get { return pk; }
                set { pk = value; }
            }
        }
        #endregion
        #region 对称钥
        public class My_k
        {
            string key; //对称钥
            public string Key 
            {
                get { return key; }
                set { key = value; }
            }
        }
        #endregion
        #region 认证数据结构
        #region 报文1
        public class Message1
        {
            string idc; //用户号
            string idt; //tgs
            string ts;  //时间戳
            public string IDC
            {
                get { return idc; }
                set { idc = value; }
            }
            public string IDT
            {
                get { return idt; }
                set { idt = value; }
            }
            public string TS
            {
                get { return ts; }
                set { ts = value; }
            }
        }
        #endregion
        #region 报文2
        public class Message2
        {
            string key; //session key
            string idt; //tgs
            string ts;  //签发时间
            string lifetime; //有效期
            Ticket t;   //票据
            public string Key
            {
                get { return key; }
                set { key = value; }
            }

            public string IDT
            {
                get { return idt; }
                set { idt = value; }
            }

            public string TS
            {
                get { return ts; }
                set { ts = value; }
            }

            public string Lifetime
            {
                get { return lifetime; }
                set { lifetime = value; }
            }

            public Ticket T
            {
                get { return t; }
                set { t = value; }
            }
        }
        #endregion
        #region 票据
        public class Ticket
        {
            string key;//Session key
            string idc;//用户标识符
            string adc;//client 地址
            string idt;//tgs
            string ts;//时间戳
            string lifetime;//有效期
            public string Key
            {
                get { return key; }
                set { key = value; }
            }
            public string IDC
            {
                get { return idc; }
                set { idc = value; }
            }
            public string ADC
            {
                get { return adc; }
                set { adc = value; }
            }
            public string IDT
            {
                get { return idt; }
                set { idt = value; }
            }
            public string TS
            {
                get { return ts; }
                set { ts = value; }
            }
            public string Lifetime
            {
                get { return lifetime; }
                set { lifetime = value; }
            }
        }
        #endregion
        #region 报文3
        public class Message3
        {
            string idv;//v的id
            Ticket t;//票据
            Authenticator ac;//认证
            public string IDV
            {
                get { return idv; }
                set { idv = value; }
            }
            public Ticket T
            {
                get { return t; }
                set { t = value; }
            }
            public Authenticator AC
            {
                get { return ac; }
                set { ac = value; }
            }
        }
        #endregion
        #region 报文4
        public class Message4
        {
            string key;//session key
            string idv;//v的id
            string ts;//时间戳
            Ticket t;//票据
            public string Key
            {
                get { return key; }
                set { key = value; }
            }
            public string IDv
            {
                get { return idv; }
                set { idv = value; }
            }
            public string TS
            {
                get { return ts; }
                set { ts = value; }
            }
            public Ticket T
            {
                get { return t; }
                set { t = value; }
            }
        }
        #endregion
        #region 认证
        public class Authenticator
        {
            string idc;//Ticket主人id
            string adc;//Ticket主人地址
            string ts;//签发时间
            public string IDC
            {
                get { return idc; }
                set { idc = value; }
            }
            public string ADC
            {
                get { return adc; }
                set { adc = value; }
            }
            public string TS
            {
                get { return ts; }
                set { ts = value; }
            }
        }
        #endregion
        #region 报文5
        public class Message5
        {
            Ticket t;//票据
            Authenticator ac;//认证
            public Ticket T
            {
                get { return t; }
                set { t = value; }
            }
            public Authenticator AC
            {
                get { return ac; }
                set { ac = value; }
            }
        }
        #endregion
        #region 报文6
        public class Message6
        {
            string ts;//时间戳
            public string TS
            {
                get { return ts; }
                set { ts = value; }
            }
        }
        #endregion
        #endregion
        #region 用户
        public class User
        {
            public string Photo { get; set; } //图片在本机的地址
            public string Uname { get; set;}//用户昵称
            public string Uid { get; set; }//用户账号
            public string Psswd { get; set; }//用户密码
            public string Name { get; set; }//用户真实姓名
            public string Sign { get; set; }//个性签名
            public string Email { get; set; }//邮箱
            public int Gender { get; set; }//性别
            public int Status { get; set; }//状态
            public int StartTime { get; set; }//注册时间
            public User()
            {
                Uid = "123456789";
                Uname = "test";
                Name = "test";
                Psswd = "0";
                Sign = "0";
                Photo = "../Image_Source/test.jpg";
                Email = Uid + "@123.com";
                Gender = 0;
                Status = 0;
                StartTime = 0;
            }

        }
        #endregion
        #region 好友
        public class Friend
        {
            User u;//用户信息
            string remark;//备注
            int startTime;//添加好友时间
            string tid;//分组信息
            public User U
            {
                get { return u; }
                set { u = value; }
            }
            public string Remark
            {
                get { return remark; }
                set { remark = value; }
            }
            public int StartTime
            {
                get { return startTime; }
                set { startTime = value; }
            }
        }
        #endregion
        #region 聊天信息
        public class chat_Message
        {
            int head;//信息种类
            string content;//信息内容
            User u;//发送方
            int time;//时间戳
            public int Head
            {
                get { return head; }
                set { head = value; }
            }
            public string Content
            {
                get { return content; }
                set { content = value; }
            }
            public User U
            {
                get { return u; }
                set { u = value; }
            }
            public int Time
            {
                get { return Time; }
                set { time = value; }
            }
        }
        #endregion
        #region 群信息
        public class Group
        {
            List<Dictionary<User, string>> list;//用户列表
            string gid;//群账号
            string photo;//群头像
            string leader;//群主账号
            string sign;//群介绍
            int startTime;//创建时间
            public List<Dictionary<User, string>> LIST
            {
                get { return list; }
                set { list = new List<Dictionary<User, string>>(value); }
            }
            public string Gid
            {
                get { return gid; }
                set { gid = value; }
            }
            public string Photo
            {
                get { return photo; }
                set { photo = value; }
            }
            public string Leader
            {
                get { return leader; }
                set { leader = value; }
            }
            public string Sign
            {
                get { return sign; }
                set { sign = value; }
            }
            public int StartTime
            {
                get { return startTime; }
                set { startTime = value; }
            }
        }
        #endregion
        #region 消息记录
        public class record_Message
        {
            List<chat_Message> messages_list;
            public List<chat_Message> Messages_list
            { 
                get { return messages_list; }
                set { messages_list = new List<chat_Message>(value); }
            }
        }
        #endregion
    }
}

File: KerberosServer/src/main/java/Json/MyStruct.java
package Json;


import java.util.*;

public class MyStruct {

    public Certificate certificate; //证书
    public My_k my_k;
    public Message1 message1;
    public Message2 message2;
    public Ticket ticket;
    public Message3 message3;
    public Message4 message4;
    public Message5 message5;
    public Message6 message6;
    public User user;
    public Friend friend;
    public Chat_Message chat_message;
    public Group group;
    public LinkedList<Group> groups;
    public Record_Message record_message;
    public LinkedList<Friend> friendlist;
/*
    static public class Friendlist{
        LinkedList<Friend> Friends;

        public LinkedList<Friend> getFriends() {
            return Friends;
        }

        public void setFriends(LinkedList<Friend> friends) {
            for(int i=0;i<friends.size();i++){
                Friends.addLast(friends.get(i));
            }
        }
    }

 */
    static public class Certificate {
         String version;//版本号
         String serial;//序列号
         String deadline;//有效日期
         String name;//主体名
         String pk;//公钥

        public String getVersion() {
            return version;
        }

        public String getSerial() {
            return serial;
        }

        public String getDeadline() {
            return deadline;
        }

        public String getName() {
            return name;
        }

        public String getPk() {
            return pk;
        }

        public void setVersion(String version) {
            this.version = version;
        }

        public void setSerial(String serial) {
            this.serial = serial;
        }

        public void setDeadline(String deadline) {
            this.deadline = deadline;
        }

        public void setName(String name) {
            this.name = name;
        }

        public void setPk(String pk) {
            this.pk = pk;
        }
    }
    //对称钥
    static public class My_k {
            String key; //对称钥

         public String getKey() {
             return key;
         }

         public  void setKey(String key) {
             this.key = key;
         }
     }
    //认证数据报文
    static public class Message1 {
         String idc; //用户号
         String idt; //tgs
         String ts;  //时间戳

         public String getIdc() {
             return idc;
         }

         public String getIdt() {
             return idt;
         }

         public String getTs() {
             return ts;
         }

         public void setIdc(String idc) {
             this.idc = idc;
         }

         public void setIdt(String idt) {
             this.idt = idt;
         }

         public void setTs(String ts) {
             this.ts = ts;
         }
     }
    static public class Message2 {
        String key; //session key
        String idt; //tgs
        String ts;  //签发时间
        String lifetime; //有效期
        String t;   //票据

        public String getKey() {
            return key;
        }

        public String getIdt() {
            return idt;
        }

        public String getTs() {
            return ts;
        }

        public String getLifetime() {
            return lifetime;
        }

        public String getT() {
            return t;
        }

        public void setKey(String key) {
            this.key = key;
        }

        public void setIdt(String idt) {
            this.idt = idt;
        }

        public void setTs(String ts) {
            this.ts = ts;
        }

        public void setLifetime(String lifetime) {
            this.lifetime = lifetime;
        }

        public void setT(String t) {
            this.t = t;
        }
    }
    static public class Ticket {
        String key;//Session key
        String idc;//用户标识符
        String adc;//client 地址
        String idt;//tgs
        String ts;//时间戳
        String lifetime;//有效期

        public String getKey() {
            return key;
        }

        public String getIdc() {
            return idc;
        }

        public String getAdc() {
            return adc;
        }

        public String getIdt() {
            return idt;
        }

        public String getTs() {
            return ts;
        }

        public String getLifetime() {
            return lifetime;
        }

        public void setKey(String key) {
            this.key = key;
        }

        public void setIdc(String idc) {
            this.idc = idc;
        }

        public void setAdc(String adc) {
            this.adc = adc;
        }

        public void setIdt(String idt) {
            this.idt = idt;
        }

        public void setTs(String ts) {
            this.ts = ts;
        }

        public void setLifetime(String lifetime) {
            this.lifetime = lifetime;
        }
    }
    static public class Message3 {
        String idv;//v的id
        String t;//票据
        String ac;//认证

        public String getIdv() {
            return idv;
        }

        public String getT() {
            return t;
        }

        public String getAc() {
            return ac;
        }

        public void setIdv(String idv) {
            this.idv = idv;
        }

        public void setT(String t) {
            this.t = t;
        }

        public void setAc(String ac) {
            this.ac = ac;
        }
    }
    static public class Message4 {
        String key;//session key
        String idv;//v的id
        String ts;//时间戳
        String t;//票据

        public String getKey() {
            return key;
        }

        public String getIdv() {
            return idv;
        }

        public String getTs() {
            return ts;
        }

        public String getT() {
            return t;
        }

        public void setKey(String key) {
            this.key = key;
        }

        public void setIdv(String idv) {
            this.idv = idv;
        }

        public void setTs(String ts) {
            this.ts = ts;
        }

        public void setT(String t) {
            this.t = t;
        }
    }
    static public class Authenticator {
        String idc;//Ticket主人id
        String adc;//Ticket主人地址
        String ts;//签发时间

        public String getIdc() {
            return idc;
        }

        public String getAdc() {
            return adc;
        }

        public String getTs() {
            return ts;
        }

        public void setIdc(String idc) {
            this.idc = idc;
        }

        public void setAdc(String adc) {
            this.adc = adc;
        }

        public void setTs(String ts) {
            this.ts = ts;
        }
    }
    static public class Message5 {
        String t;//票据
        String ac;//认证

        public String getT() {
            return t;
        }

        public String getAc() {
            return ac;
        }

        public void setT(String t) {
            this.t = t;
        }

        public void setAc(String ac) {
            this.ac = ac;
        }
    }
    static public class Message6 {
        String ts;//时间戳

        public String getTs() {
            return ts;
        }

        public void setTs(String ts) {
            this.ts = ts;
        }
    }
    static public class User
    {
        String Photo ; //图片在本机的地址
        String Uname ;//用户昵称
        String Uid ;//用户账号
        String Psswd ;//用户密码
        String Name;//用户真实姓名
        String Sign ;//个性签名
        String Email;//邮箱
        int Gender ;//性别
        int Status ;//状态
        long StartTime;//注册时间

        public String getPhoto() {
            return Photo;
        }

        public String getUname() {
            return Uname;
        }

        public String getUid() {
            return Uid;
        }

        public String getPsswd() {
            return Psswd;
        }

        public String getName() {
            return Name;
        }

        public String getSign() {
            return Sign;
        }

        public String getEmail() {
            return Email;
        }

        public int getGender() {
            return Gender;
        }

        public int getStatus() {
            return Status;
        }

        public long getStartTime() {
            return StartTime;
        }

        public void setPhoto(String photo) {
            Photo = photo;
        }

        public void setUname(String uname) {
            Uname = uname;
        }

        public void setUid(String uid) {
            Uid = uid;
        }

        public void setPsswd(String psswd) {
            Psswd = psswd;
        }

        public void setName(String name) {
            Name = name;
        }

        public void setSign(String sign) {
            Sign = sign;
        }

        public void setEmail(String email) {
            Email = email;
        }

        public void setGender(int gender) {
            Gender = gender;
        }

        public void setStatus(int status) {
            Status = status;
        }

        public void setStartTime(long startTime) {
            StartTime = startTime;
        }
    }
    static public class Friend {
        User u;//用户信息
        String remark;//备注
        int startTime;//添加好友时间
        String tid;//分组信息

        public User getU() {
            return u;
        }

        public void setU(User u) {
            this.u= new MyStruct.User();
            this.u.setUname(u.getUname());
            this.u.setUid(u.getUid());
            this.u.setStatus(u.getStatus());
            this.u.setStartTime(u.getStartTime());
            this.u.setSign(u.getSign());
            this.u.setPhoto(u.getPhoto());
            this.u.setName(u.getName());
            this.u.setGender(u.getGender());
            this.u.setEmail(u.getEmail());
        }

        public String getRemark() {
            return remark;
        }

        public void setRemark(String remark) {
            this.remark = remark;
        }

        public int getStartTime() {
            return startTime;
        }

        public void setStartTime(int startTime) {
            this.startTime = startTime;
        }

        public String getTid() {
            return tid;
        }

        public void setTid(String tid) {
            this.tid = tid;
        }
    }
    static public class Chat_Message {
        String content;//信息内容
        User u;//发送方
        long time;//时间戳

        

        public String getContent() {
            return content;
        }

        public void setContent(String content) {
            this.content = content;
        }

        public User getU() {
            return u;
        }

        public void setU(User u) {
            this.u = u;
        }

        public long getTime() {
            return time;
        }

        public void setTime(long time) {
            this.time = time;
        }
    }
    static public class Group {
        List<User> list=new LinkedList<User>();//用户列表
        String gid;//群账号
        String gname;//群名称
        String photo;//群头像
        String leader;//群主账号
        String sign;//群介绍
        long startTime;//创建时间

        public String getGname() {
            return gname;
        }

        public void setGname(String gname) {
            this.gname = gname;
        }

        public List<User> getList() {
            return list;
        }

        public void setList(List<User> list) {
            this.list = list;
        }

        public String getGid() {
            return gid;
        }

        public void setGid(String gid) {
            this.gid = gid;
        }

        public String getPhoto() {
            return photo;
        }

        public void setPhoto(String photo) {
            this.photo = photo;
        }

        public String getLeader() {
            return leader;
        }

        public void setLeader(String leader) {
            this.leader = leader;
        }

        public String getSign() {
            return sign;
        }

        public void setSign(String sign) {
            this.sign = sign;
        }

        public long getStartTime() {
            return startTime;
        }

        public void setStartTime(long startTime) {
            this.startTime = startTime;
        }
    }
    static public class Record_Message {
        List<Chat_Message> messages_list;

        public List<Chat_Message> getMessages_list() {
            return messages_list;
        }

        public void setMessages_list(List<Chat_Message> messages_list) {
            this.messages_list = messages_list;
        }
    }
}
